### PR TITLE
Upload: Dropzone allows now multiple files

### PR DESCRIPTION
**Pull request by mstipcevic**

### CODE REVIEW

File: solute/epfl/components/upload/upload.html [hunk 1]
```diff
@@ -34,7 +34,14 @@
                     {% if compo.show_remove_icon %}
                         <i class="fa fa-times fa-lg color-danger epfl-upload-remove-icon"></i>
                     {% endif %}
-                    <img src="{{ compo.value }}" class="epfl-upload-image" data-cid="{{ compo.cid }}"/>
+
+                  {% if compo.value is string %}
+                      <img src="{{ compo.value }}" class="epfl-upload-image" data-cid="{{ compo.cid }}"/>
+                      {% else %}
+                        {% for file in compo.value %}
```
Comment (KingWarin) [inline solute/epfl/components/upload/upload.html]: None check is missing

File: solute/epfl/components/upload/upload.html [hunk 2]
```diff
@@ -34,7 +34,14 @@
                     {% if compo.show_remove_icon %}
                         <i class="fa fa-times fa-lg color-danger epfl-upload-remove-icon"></i>
                     {% endif %}
-                    <img src="{{ compo.value }}" class="epfl-upload-image" data-cid="{{ compo.cid }}"/>
+
```
Comment (KingWarin) [inline solute/epfl/components/upload/upload.html]: The whole if/else below looks pretty messy (watch for indentation)

File: solute/epfl/components/upload/static/upload.js [hunk 3]
```diff
@@ -128,88 +128,180 @@ epfl.Upload.prototype.read_file = function (file, callback) {
...
+            if (!file_data[i].valid) {
+                continue;
+            }
+            file_infos.push({
+                size: file_data[i].file_size,
+                type: file_data[i].file_type,
+                name: file_data[i].file_name,
+                is_image: file_data[i].file_is_img,
+                image_width: file_data[i].file_img_width,
+                image_height: file_data[i].file_img_height,
```
Comment (KingWarin) [inline solute/epfl/components/upload/static/upload.js]: Remove dangling ',' and blankline

File: solute/epfl/components/upload/static/upload.js [hunk 4]
```diff
@@ -128,88 +128,180 @@ epfl.Upload.prototype.read_file = function (file, callback) {
...
-        obj.upload_file(this, files[0])
-    });
-};
 
-epfl.Upload.prototype.upload_file = function (reader, file) {
-    var obj = this;
+    var validate_files = function () {
+        //validate all files in file_data
+        for (var i = 0; i < file_data.length; i++) {
+
```
Comment (KingWarin) [inline solute/epfl/components/upload/static/upload.js]: Blank line?

File: solute/epfl/components/upload/static/upload.js [hunk 5]
```diff
@@ -128,88 +128,180 @@ epfl.Upload.prototype.read_file = function (file, callback) {
...
             obj.is_async_uploading = false;
             spinner.remove();
+            //TODO: make result visible
             obj.handle_drop_url(data);
         });
-    } else {
-        obj.change(reader.result);
+    };
+
+    //read all filesm extract their data and call next functions
```
Comment (KingWarin) [inline solute/epfl/components/upload/static/upload.js]: typo?

File: solute/epfl/components/upload/upload.html (hunk 1 above)
Comment (mstipcevic) [inline solute/epfl/components/upload/upload.html]: none check above

**Pull request merged**